### PR TITLE
telemetry: add tips for agent mode

### PR DESCRIPTION
**Pull request by zirain**
fixes: https://github.com/istio/istio/issues/43575

cc @lei-tang

### CODE REVIEW

File: content/en/docs/tasks/observability/telemetry/index.md [hunk 1]
```diff
@@ -73,6 +73,10 @@ will be used when the `Telemetry` resources do not specify a provider.
 If you're using [Sidecar](/docs/reference/config/networking/sidecar/) configuration, do not forget to add provider's service.
 {{< /tip >}}
 
+{{< tip >}}
+Provider does not support `$(HOST_IP)`, If you're running collector in agent mode, you can [using service internal traffic policy](https://kubernetes.io/docs/concepts/services-networking/service-traffic-policy/#using-service-internal-traffic-policy) to `Local` for better performance.
```
Comment (johnzheng1975) [inline content/en/docs/tasks/observability/telemetry/index.md:77]:
```suggestion
Provider does not support `$(HOST_IP)`. If you're running collector in agent mode, you can [use service internal traffic policy](https://kubernetes.io/docs/concepts/services-networking/service-traffic-policy/#using-service-internal-traffic-policy), set internalTrafficPolicy to `Local` for better performance.
```

**Review by johnzheng1975 — APPROVED** [review]

File: content/en/docs/tasks/observability/telemetry/index.md [hunk 2]
```diff
@@ -73,6 +73,10 @@ will be used when the `Telemetry` resources do not specify a provider.
 If you're using [Sidecar](/docs/reference/config/networking/sidecar/) configuration, do not forget to add provider's service.
 {{< /tip >}}
 
+{{< tip >}}
+Provider does not support `$(HOST_IP)`. If you're running collector in agent mode, you can [use service internal traffic policy](https://kubernetes.io/docs/concepts/services-networking/service-traffic-policy/#using-service-internal-traffic-policy), set `InternalTrafficPolicy` to `Local` for better performance.
```
Comment (kfaseela) [inline content/en/docs/tasks/observability/telemetry/index.md:77]: which provider?

Comment (zirain) [reply content/en/docs/tasks/observability/telemetry/index.md:77]: > The Telemetry API uses the concept of providers to indicate the protocol or type of integration to use. Providers can be configured in [`MeshConfig`](/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig-ExtensionProvider).

Comment (kfaseela) [reply content/en/docs/tasks/observability/telemetry/index.md:77]: i saw different providers above, so was wondering if the tip is for any specific provider or all of them in general.

Comment (kfaseela) [inline content/en/docs/tasks/observability/telemetry/index.md:77]:
```suggestion
Providers do not support `$(HOST_IP)`. If you're running collector in agent mode, you can use [service internal traffic policy](https://kubernetes.io/docs/concepts/services-networking/service-traffic-policy/#using-service-internal-traffic-policy), and set `InternalTrafficPolicy` to `Local` for better performance.
```

**Pull request merged**